julia/kernels/bb_template.cxx

```diff
@@ -275,15 +272,15 @@ cudaEvent_t cuda{{{kernel_name}}}::execute(cudaPipelineState& /*pipestate*/, con
     if (err != CUDA_SUCCESS) {
         const char* errStr;
         cuGetErrorString(err, &errStr);
-        ERROR("cuLaunchKernel: Error number: {}: {}", err, errStr);
+        ERROR("cuLaunchKernel: Error number: {}: {}", (int)err, errStr);
     }
 
     // Copy results back to host memory
     // TODO: Skip this for performance
     {{#kernel_arguments}}
         {{^hasbuffer}}
             {{#isoutput}}
-                CHECK_CUDA_ERROR(cudaMemcpyAsync({{{name}}}_host.at(gpu_frame_index).data(),
```

Comment:
This is similar to my comment above, but this code handles output buffers that are hidden from Kotekan. These are the `info` output arguments for the kernel. I don't know if kernels can now be launched in an overlapping manner, and if so, we would need to make sure that these `info` output arguments are separate for each call. If the kernels remain sufficiently serialized then your code is correct.